includes/class-sensei-quiz.php

```diff
@@ -159,27 +162,32 @@ public function get_lesson_id( $quiz_id ) {
 	 * This function hooks into the quiz page and accepts the answer form save post.
 	 *
 	 * @since 1.7.3
-	 * @return bool $saved;
 	 */
 	public function user_save_quiz_answers_listener() {
 
 		if ( ! isset( $_POST['quiz_save'] )
-			|| ! isset( $_POST['sensei_question'] )
 			|| empty( $_POST['sensei_question'] )
 			|| ! isset( $_POST['woothemes_sensei_save_quiz_nonce'] )
 			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Do not change the nonce.
 			|| ! wp_verify_nonce( wp_unslash( $_POST['woothemes_sensei_save_quiz_nonce'] ), 'woothemes_sensei_save_quiz_nonce' ) ) {
 			return;
 		}
 
-		global $post;
-		$lesson_id    = $this->get_lesson_id( $post->ID );
-		$quiz_answers = $this->merge_quiz_answers_with_questions_asked( $_POST, $post->ID );
+		$quiz_id   = get_the_ID();
+		$lesson_id = $this->get_lesson_id( $quiz_id );
+		$user_id   = get_current_user_id();
+
+		$answers = $this->parse_form_answers(
+			$_POST['sensei_question'], // phpcs:ignore WordPress.Security.ValidatedSanitizedInput
```

Comment:
It would be helpful here to include a brief description as to why we can safely ignore this. Same goes for all other instances of this.

Reply:
I've made some improvements in 27bcb900a06b570d49555515cf234d1caddeec3f, also added the description.

Reply:
Maybe at some point, we should consider using `htmlspecialchars` to improve security. I've made a quick test, but it looks like the multi-line question type is breaking.

I'm not sure about using `sanitize_text(area)_field`, because it strips all HTML tags and some quizzes might require answers with HTML.

Also, I'm not sure what this change might do to third-party code. 🙂

Reply:
> I'm not sure what this change might do to third-party code.

😄 Ha ha. Yeah, might be worth it at some point. I have seen some sites who use Sensei for programming courses, and they could conceivably ask students to write HTML in quiz questions.